### ISSUE TITLE
Importing with webpack 3.10 results in build errors because of errorneous node detection

### ISSUE THREAD

**Issue by VanTanev** [pubkey/broadcast-channel#2]
```
ERROR in ./node_modules/broadcast-channel/dist/es/methods/node.js
Module not found: Error: Can't resolve 'fs' in '/home/ivan/dev/MAP3/scar-front-end/node_modules/broadcast-channel/dist/es/methods'
 @ ./node_modules/broadcast-channel/dist/es/methods/node.js 9:0-25
 @ ./node_modules/broadcast-channel/dist/es/method-chooser.js
 @ ./node_modules/broadcast-channel/dist/es/index.js
 @ ./app/js/app.js
 @ multi (webpack)-dev-server/client?http://localhost:8087 webpack/hot/dev-server ./app/js/app.js
```

A workaround to get a working build is to add the following to webpack.config:
```
{
  node: {
    fs: 'empty',
    net: 'empty',
  }
}
```

It seems that webpack bundles the node method, regardless of the fact that it's never required, because the decision if it should be required is a runtime decision from webpack's perspective. The `if(isNode)` conditional cannot be optimized away. This wastes 36kb of the total 52kb of broadcast-channel bundle, just to import node method and related node libs that can run in the browser.

I haven't yet figured out the exact solution, but I think it should be possible to use the `browser` field in `package.json` to specify the node method as an empty module for browser envs.

**Comment by rianmcguire** [comment]
I ran into this too. My workaround was to use the webpack.IgnorePlugin:
```
new webpack.IgnorePlugin(/node\.js$/, /broadcast-channel/)
```

That avoids having the node code in the bundle, but ideally they would be no extra configuration required.

**Comment by pubkey** [comment]
I added a test for the webpack-build and included the proposed method via `browser`-field in the `package.json`.
Please test if this works for you.

**Comment by VanTanev** [comment]
@pubkey Tested the change, it seems to work as expected!

I wonder if we should have a guard around `METHODS.push(NodeMethod);`, because now in a webpack env `REQUIRE_FUN('./methods/node.js')` returns an empty object `{}`. Normally, we should not hit that, but it's worth noting that we now have two independent ways of checking for a node - `node-detect` and in a webpack env, webpack's dummy module replacement. Maybe just an extra guard:
```
if (NodeMethod.type === 'node') {
  METHODS.push(NodeMethod);
}
```

**Comment by pubkey** [comment]
@VanTanev I [did this](https://github.com/pubkey/broadcast-channel/commit/9900b741552b744fad790d6bb3464b925c40d73a#diff-7393277c0aec4f81882bccabeb18d352L40). Which seems strange.
I will switch to your proposal.

**Comment by pubkey** [comment]
I added the change and released `1.2.1` to npm. This should be fixed now.

**Comment by VanTanev** [comment]
@pubkey Thanks a lot for the quick response and fix, as well for the excellent module!

All the best!
